doc/distributed_python/usage.rst

```diff
@@ -1,2 +1,33 @@
 Creating and running UDFs
--------------------------
\ No newline at end of file
+-------------------------
+
+For best performance, we recommend that you create a script using CREATE SCRIPT and then use this script within a SELECT statement. Embedding the script within SQL in this way provides the best performance and scalability.
+
+In the CREATE SCRIPT command, you must define the type of input and output values.
+There are two types of UDF inputs (set and scalar) and two types of UDF outputs (returns and emits). 
+These can be combined as needed to suite your use case.
+
+- Input values
+
+    - **SCALAR** Specifies that the script processes single input rows. The code is therefore called once per input row.
+
+    - **SET** Specifies that the processing refers to a set of input values. Within the code, you can iterate through those values.
+
+- Output values
+
+    - **RETURNS** Specifies that the script returns a single value.
+
+    - **EMITS** Specifies that the script can create (emit) multiple result rows (tuples).
+
+You can define the data types of input and output parameters to specify the conversion between internal data types and the database SQL data types. If you do not specify the data types, the script has to handle that dynamically.
```

Comment:
I think you have to specify the type unless you use a variadic script. For the output parameters you still need to specify the type, you just do it in a different place.